app/warrior/knight.py

```diff
@@ -0,0 +1,36 @@
+class Knight:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: list | None,
+            weapon: dict | None,
+            potion: dict | None
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = 0
+
+    def apply_armour(self) -> None:
+        for arm in self.armour:
+            self.protection += arm["protection"]
+
+    def apply_weapon(self) -> None:
+        self.power += self.weapon["power"]
```

Comment:
The `apply_weapon` method does not check if `self.weapon` is `None` before accessing it. This could lead to a `TypeError` if `self.weapon` is `None`. Consider adding a check to ensure `self.weapon` is not `None` before accessing its 'power' key.